lib/formulator.ex

```diff
@@ -2,31 +2,80 @@ defmodule Formulator do
   use Phoenix.HTML
   alias Formulator.HtmlError
 
-  def input(form, field, options \\ []) do
+  @doc  """
+  Returns an html input with an associated label.
+
+  Passing `label: false` into the options will return the input without a label.
+  In that case it will add an `aria-label` attribute to the input to improve
+  accessibility.
+
+  ## Examples
+
+  Basic input:
+      <%= input form, :email %>
+      #=> <label for="user_email">Email</label>
+      #=> <input id="user_name" name="user[email]" type="text" value="">
+
+  Without a label:
+      <%= input form, :email, label: false %>
+      #=> <input id="user_name" name="user[email]" aria-label="email" type="text" value="">
+  """
+
+  @spec input(Phoenix.HTML.Form.t, atom, []) :: binary
+  def input(form, field, options \\ [])
+
+  def input(form, field, options) do
+    cond do
+      {:label, false} in options -> input_without_label(form, field, options)
+      true -> input_with_label(form, field, options)
+    end
+  end
+
+  defp input_without_label(form, field, options) do
+    options = options ++ build_aria_label(field)
+    error = html_error(form, field)
+    [
+      build_input(form, field, options, error)
+    ] ++ error.html
+  end
+
+  defp input_with_label(form, field, options) do
     error = html_error(form, field)
     build_html(form, field, options, error)
   end
 
+  defp build_aria_label(field) do
+    ["aria-label": field |> to_string |> String.replace("_", " ")]
```

Comment:
This is a little terse. Thoughts on extracting `field |> to_string |> String.replace("_", " ")` to a variable so it's easy to see that the return is `["aria-label": some_value]` at a glance?